### PR TITLE
Asciidoc README support

### PR DESCRIPTION
**Pull request by Mizuho32**
Asciidoc format README support. https://github.com/gogits/gogs/issues/2097

gogs-adoc2

```adoc
= Hello

== First

[red]#*test*#
http://asciidoctor.org[*Asciidoctor*]

[TIP]
====
Hoge
====
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,6 +5,10 @@ DATA_FILES := $(shell find conf | sed 's/ /\\ /g')
 LESS_FILES := $(wildcard public/less/gogs.less public/less/_*.less)
 GENERATED  := pkg/bindata/bindata.go public/css/gogs.css
 
+ADOCJS		 := node_modules/asciidoctor.js/dist/asciidoctor.min.js
```
Comment (Unknwon) [inline Makefile]: Do not check in node_modules, only import final .min.js and .min.css under `public/plugins` directory. Look for plugins already under that directory for example.

File: Makefile [hunk 2]
```diff
@@ -19,6 +23,10 @@ GOVET = go tool vet -composites=false -methods=false -structtags=false
 
 .IGNORE: public/css/gogs.css
 
+adocjsupdate: $(NODE_MODULES)
```
Comment (Unknwon) [inline Makefile]: No need to update in makefile, download a specific version is all we need.

**Comment by Mizuho32** [comment]
I removed asciidoctor update code from Makefile and moved asciidoctor code to `plugins/asciidoctor-1.5.7/`

**Comment by pjanx** [comment]
Does this fix links to images?
```
image::picture.png[align="center"]
```
Something like [in Markdown](https://github.com/gogs/gogs/blob/master/pkg/markup/markup.go#L175). I haven't tried your changes first-hand to see if it does but it seems to me that you've omitted this handling, which would be a big shame.

By the way https://github.com/bytesparadise/libasciidoc is getting somewhere.

**Comment by Mizuho32** [comment]
For the images on the web, the answer is Yes.
gogs_adoc_image

source:
```adoc
= Hello

== Asciidoctor

image::https://user-images.githubusercontent.com/79351/32431984-fb09a580-c292-11e7-9a10-553d46dba2b3.png[align="center"]

== First

[red]#*test*#
http://asciidoctor.org[*Asciidoctor*]

[TIP]
====
Hoge
====
```

The script Asciidoctor.js I used in this PR, is a browser side script and converts the adoc document to the html at the browser client side on load.
So, I don't know the detail of the markup.go and the function `wrapImgWithLink`, it doesn't need such kind of implementation and is independent of the server side implementation, languages.

For the pictures on the server local (e.g. `/home/hoge/test.png`),  it is impossible to access.

**Comment by pjanx** [comment]
I'm talking about linking to pictures that are in the same repository as the viewed Asciidoc file.

**Comment by Mizuho32** [comment]
If those pictures have the urls like `https://user-images.githubusercontent.com/79351/32431984-fb09a580-c292-11e7-9a10-553d46dba2b3.png`, it is possible to link but maybe you want to link the pictures in the simple, relative path style like `image::picture.png[align="center"]` so, it is not possible.

**Comment by pjanx** [comment]
Okay, at least that's been answered. But it is very possible, not solving it is a bug:
* you can always post-process the output in JS for one,
* or set a default value for the `imagesdir` attribute
   
  > If the image target is an absolute path or URI, the imagesdir prefix is not added to the path.

**Comment by Mizuho32** [comment]
I guess, if the pictures' ulrs have the rules like below, post-processing or attributes like `imagesdir` will make sense.
```bash
├── README.adoc
├── chakku.png
└── img
    └── poi.png
```
* `http://raw.poioichakku.com/chakku.png` for `chakku.png`
* `http://raw.poioichakku.com/img/poi.png` for `poi.png`

For the case that the pictures' ulrs are like `https://user-images.githubusercontent.com/79351/32431984-fb09a580-c292-11e7-9a10-553d46dba2b3.png`, including random string, I have no idea....

Fortunately, gogs's url rule is the first one(?), so I'll try (but I have no time to do now 😭)

**Pull request closed without merging**